.github/workflows/_snap-release.yaml

```diff
@@ -0,0 +1,79 @@
+on:
+  workflow_call:
+    inputs:
+      track:
+        description: |
+          The track to which charm should be released. This input supports a
+          comma-separated tracks in string format. The channels will be a list of
+          track/risk-level combinations. By default, the snap will be released to
+          channel(s) in the "latest" track.
+          Examples:
+          - "latest"
+          - "latest,test,2.9"
+        required: false
+        type: string
+        default: latest
+      risk-level:
+        description: |
+          The risk level to which charm should be released. Valid inputs are one or
+          a combination (in comma-separated string format) of: "edge", "beta",
+          "candidate", "stable". The channels will be a list of track/risk-level
+          combinations. By default, the snap will be released to channel(s) with
+          "edge" risk level.
+          Examples:
+          - "edge"
+          - "edge,candidate"
+        required: false
+        type: string
+        default: edge
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: Publish Snap
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: snap-artifact
+      - id: get-snap-name
+        run: echo "SNAP_FILE=$(ls *.snap)" >> $GITHUB_OUTPUT
+      - name: Generate a comma separated list of channels to release the snap to
+        id: get-channels
+        env:
+          TRACKS: ${{ inputs.track }}
+          RISKLEVELS: ${{ inputs.risk-level }}
+        run: |
+          # split comma-separated TRACKS and RISKLEVELS to arrays
```

Comment:
if you're already using bash arrays, you could consider making this marginally simpler by setting CHANNELS to an empty array, appending `"$i/$j"` as you go, and then finally turning it into a string with something like `echo "${CHANNELS[@]}" | tr ' ' ','` (avoiding the "if CHANNELS is empty else" conditional)